Foxtrot/Driver/Rewriting/PostRewriteChecker.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Compiler;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Contracts.Foxtrot
+{
+    public sealed class PostRewriteChecker : Inspector
+    {
+        private Member currentMember;
+        private readonly Action<CompilerError> handleError;
+
+        public PostRewriteChecker(Action<CompilerError> handleError)
+        {
+            this.handleError = handleError;
+        }
+
+        public override void VisitProperty(Property property)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
+            {
+                this.currentMember = property;
+                exchanged = true;
+            }
+
+            base.VisitProperty(property);
+
+            if (exchanged)
+            {
+                this.currentMember = null;
+            }
+        }
+
+        public override void VisitEvent(Event evnt)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
```

Comment:
I'm not 100% sure what `VisitEvent` and `VisitProperty` methods are for... 

I assume that this behavior is clear if you're stepping through the debugger, but reading this code is a bit confusing...